presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -174,6 +200,9 @@ public HashAggregationOperatorFactory(
             this.groupByTypes = ImmutableList.copyOf(groupByTypes);
             this.groupByChannels = ImmutableList.copyOf(groupByChannels);
             this.globalAggregationGroupIds = ImmutableList.copyOf(globalAggregationGroupIds);
+            this.sourceTypes = ImmutableList.copyOf(requireNonNull(sourceTypes, "orderByTypes is null"));
```

Comment:
ditto. Please add check state that when there are orderings then step is `SINGLE`